### DISCUSSION TITLE
play framework java agent configuration

### DISCUSSION THREAD

**Discussion by someone-1** [apache/skywalking#10381] [Q&A]
I was trying to integrate the java agent with 
this play application
https://github.com/playframework/play-samples/tree/2.8.x/play-java-rest-api-example 
the entry spans are not captured, I only see the exit spans

steps to setup the application
clone this repo and go to this directory
https://github.com/playframework/play-samples/tree/2.8.x/play-java-rest-api-example 

run 
`sbt dist`

unzip the generated zip 
`unzip /target/universal/play-java-rest-api-example-2.8.x.zip`

to run the project 
`. /target/universal/play-java-rest-api-example-2.8.x/bin/play-java-rest-api-example`

this is to run the play project.

now this is steps followed to integrate the java agent

`export JAVA_OPTS="$JAVA_OPTS -javaagent:path/to/java/agent.jar"`
and restart the project

pls let me know if I missed something

edit:  using java 8

followed the same steps for different application
https://github.com/playframework/play-samples/tree/2.8.x/play-java-hello-world-tutorial
its working for this project,
able to see entry spans, other metrics

**Comment by wu-sheng** [comment]
The play framework was contributed by @aiyanbo and verified through this demo application

https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/play-scenario

SkyWalking plugins are from hundreds of people. You are better to compare your use case witv existing test demo. Maybe some cases are not covered.

**Comment by someone-1** [comment]
FYI, I was debugging, this method is not getting invoked  
in the [play-java-rest-api-example](https://github.com/playframework/play-samples/tree/2.8.x/play-java-rest-api-example) example
https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/play-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/play/v2x/HttpFiltersInterceptor.java#L45

**Reply by wu-sheng** [reply]
Then you should check the interceptor point of `HttpFiltersInterceptor`? Maybe it doesn't match the current codes anymore?

**Reply by wu-sheng** [reply]
This is the key

https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/play-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/play/v2x/define/Play2xInstrumentation.java

**Reply by someone-1** [reply]
https://www.playframework.com/documentation/2.8.x/Filters#Default-Filters
> Play now comes with a default set of enabled filters, defined through configuration. If the property play.http.filters is null, then the default is now play.api.http.EnabledFilters

https://github.com/playframework/play-samples/blob/2.8.x/play-java-rest-api-example/conf/application.conf#L25
after commenting this line its working

**Reply by wu-sheng** [reply]
Sorry, I don't know play framework. I can't tell what means about this filter.
